cmd/kubeapps-apis/server/packages.go

```diff
@@ -47,14 +49,32 @@ func (s packagesServer) GetAvailablePackageSummaries(ctx context.Context, reques
 
 	log.Infof("+core GetAvailablePackageSummaries %s", contextMsg)
 
+	pageOffset, err := pageOffsetFromPageToken(request.GetPaginationOptions().GetPageToken())
+	pageSize := request.GetPaginationOptions().GetPageSize()
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to intepret page token %q: %v", request.GetPaginationOptions().GetPageToken(), err)
+	}
+
+	// TODO(agamez): temporarily fetching all the results (size=0) and then paginate them
+	// ideally, paginate each plugin request and then aggregate results.
+	requestN := request
+	requestN.PaginationOptions = &packages.PaginationOptions{
+		PageToken: "0",
+		PageSize:  0,
+	}
+
 	pkgs := []*packages.AvailablePackageSummary{}
+	categories := []string{}
+
 	// TODO: We can do these in parallel in separate go routines.
 	for _, p := range s.plugins {
-		response, err := p.server.GetAvailablePackageSummaries(ctx, request)
+		response, err := p.server.GetAvailablePackageSummaries(ctx, requestN)
```

Comment:
Good idea, at least we prevent unnecessary plugin requests. Done. But certainly we have to come up with a proper way of pagination :S

Reply:
I think we already have one in mind, we just need to update the existing pagination (throughout) to use the number of items as the offset, rather than the number of pages.